src/js/components/useProgressRing.ts

```diff
@@ -33,19 +34,38 @@ export interface ProgressRingReturn {
   error?: Error
 }
 
-// Not testable because it manipulates SVG elements, unsupported by JSDom
-const useProgressRing = (element: HTMLElement | null): ProgressRingReturn => {
-  if (element) {
-    const circle = element.querySelector<SVGCircleElement>(ProgressRingMap.checkout.circle);
+export const TextType = {
+  enum: 'enum',
+  percent: 'percent',
+};
+
+export interface ProgressRingOptions {
+  steps: number;
+  text: keyof typeof TextType;
+}
```

Comment:
```suggestion
export enum TextType {
  enum: 'enum',
  percent: 'percent',
};

export interface ProgressRingOptions {
  steps: number;
  text: TextType;
}
```

Does it works like this?

Reply:
In modern TypeScript, you may not need an enum when an object with as const could suffice:
https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums

but here i forget to add `as const`  to the TextType object!

Reply:
But using an enum for such a thing looks like a good practice on different projects

Reply:
I've tested, there is a problem with eslint when using enum here.
I think it's a bug in eslint:
https://stackoverflow.com/questions/63961803/eslint-says-all-enums-in-typescript-app-are-already-declared-in-the-upper-scope

Reply:
Isn't it related to the enum keyword that should be named differently?

Reply:
Using object with `as const` is safer:
![Screenshot from 2022-03-29 20-35-33](https://user-images.githubusercontent.com/85633460/160655999-328388c7-2c92-4b95-a7b4-5f6742f800e1.png)